### PR TITLE
Considering receiverType on function test creation issue-2282

### PR DESCRIPTION
**Pull request by marcel-basel**
The receiver type for a method is now considered. So, if there are >= 1 method names in the same file, then clicking "generate unit tests for function" creates test functions only for the selected method.

funcName like "(*Point).Move" will results in "gotests -w -only ^PointMove$ calls.

### CODE REVIEW

File: src/goGenerateTests.ts [hunk 1]
```diff
@@ -98,8 +98,16 @@ export async function generateTestCurrentFunction(): Promise<boolean> {
 		return Promise.resolve(false);
 	}
 	let funcName = currentFunction.name;
-	if (funcName.includes('.')) {
-		funcName = funcName.split('.')[1];
+	if (funcName.includes('.')) {  // receiverType specified
+		let funcNameParts = funcName.match(/^\(\*?(.*)\)\.(.*)$/);
```
Comment (ramya-rao-a) [inline src/goGenerateTests.ts]: If we are going to use a regular expression, then the `if (funcName.includes('.'))` check is redundant right?

Do we foresee the function name to include `.` in a non receiverType case?

Comment (marcel-basel) [reply src/goGenerateTests.ts]: Hi Ramya, thanks for your fantastic work on the extension!

I removed the `if (funcName.includes('.'))` block. I don't see an other case where the function name would include `.` Do you?

Comment (ramya-rao-a) [reply src/goGenerateTests.ts]: Nope, change looks good. I have pushed a commit to fix the linting errors, we are good to merge now

**Comment by marcel-basel** [comment]
Thanks @ramya-rao-a, do you consider #2282 as closable?

**Comment by ramya-rao-a** [comment]
I'll be closing it once I release an update with the fix.

**Pull request merged**